.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,63 @@
+name: e2e
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  e2e:
+    name: Run acceptance tests against OpenStack ${{ matrix.name }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "dalmatian"
+            openstack_version: "stable/2024.2"
+            ubuntu_version: "22.04"
+          - name: "caracal"
+            openstack_version: "stable/2024.1"
+            ubuntu_version: "22.04"
+    env:
+      image_tag: virtual-registry.k-orc.cloud/ci:commit-${GITHUB_SHA::7}
+
+    runs-on: ubuntu-${{ matrix.ubuntu_version }}
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Deploy devstack
+      uses: EmilienM/devstack-action@40c77372dbc135a17adc877eb77fc226a134305c
+      with:
+        enable_workaround_docker_io: 'false'
+        branch: ${{ matrix.openstack_version }}
+
+    - name: Deploy a Kind Cluster
+      uses: helm/kind-action@dda0770415bac9fc20092cacbc54aa298604d140
+      with:
+        cluster_name: orc
+
+    - name: Build and push a container image to Kind
+      run: |
+        docker build -t ${{ env.image_tag }} .
+        kind load docker-image ${{ env.image_tag }} ${{ env.image_tag }} --name orc
+
+    - name: Deploy orc
+      run: |
+        kubectl config use-context kind-orc
+        make deploy IMG=${{ env.image_tag }}
+
+    - name: Apply simple-server
+      run: |
+        cp /etc/openstack/clouds.yaml examples/credentials/
+        make --directory=examples/dev-settings/
+        cd examples/public-server
+        kubectl kustomize . | kubectl apply -f - --server-side
+        kubectl wait --timeout=10m --for=condition=ready server ${USER}-public-server
+
+    - name: Inspect the server
+      run: |
+        openstack server show "$(kubectl get server ${USER}-public-server -o jsonpath='{.status.resource.id}')"
+      env:
+        OS_CLOUD: devstack
```

Comment:
Although I am ok with this as an extremely rough PoC, as an actual testing strategy this pattern would be a hard no from me. Our tests must be runnable anywhere, and must not be tied to a particular CI. Specifically, it must be possible to run tests locally.

Ideally this action definition would stop after `deploy devstack` and everything subsequent is defined in a make action, including standing up a kind cluster and the build step.

I'm ok with it for now, but I'll probably NAK it the first time anybody tries to add anything new to it.